src/components/chat-components/RelevantNotes.tsx

```diff
@@ -0,0 +1,261 @@
+  const navigateToNote = (notePath: string) => {
+    const file = app.vault.getAbstractFileByPath(notePath);
+    if (file instanceof TFile) {
+      app.workspace.getLeaf(false).openFile(file);
+    }
+  };
+  const addToChat = (prompt: string) => {
+    onInsertToChat(`[[${prompt}]]`);
+  };
+  return (
+    <Card className={cn("w-full bg-secondary border border-solid border-border", className)}>
+      <Collapsible open={isOpen} onOpenChange={setIsOpen}>
+        <CardHeader className="p-4 border-b">
+          <CardTitle className="flex justify-between items-center">
+            <div className="flex gap-2 items-center">
+              <span>Relevant Notes</span>
+              {!hasIndex && (
+                <Tooltip>
+                  <TooltipTrigger asChild>
+                    <TriangleAlert className="size-4 text-warning" />
+                  </TooltipTrigger>
+                  <TooltipContent>Note has not been indexed</TooltipContent>
+                </Tooltip>
+              )}
+            </div>
+            {relevantNotes.length > 0 && (
+              <CollapsibleTrigger asChild>
+                <button className="size-6 p-0 !bg-transparent border-none !shadow-none hover:!bg-interactive-hover">
+                  {isOpen ? <ChevronUp className="size-4" /> : <ChevronDown className="size-4" />}
+                </button>
+              </CollapsibleTrigger>
+            )}
+          </CardTitle>
+          {relevantNotes.length === 0 && (
+            <CardDescription className="flex flex-wrap gap-x-2 gap-y-1 max-h-12 overflow-y-hidden">
+              <span className="text-xs text-muted">No relevant notes found</span>
+            </CardDescription>
+          )}
+          {!isOpen && relevantNotes.length > 0 && (
+            <CardDescription className="flex flex-wrap gap-x-2 gap-y-1 max-h-12 overflow-y-hidden">
+              {relevantNotes.map((note) => (
+                <RelevantNotePopover
+                  note={note}
+                  onAddToChat={() => addToChat(note.document.title)}
+                  onNavigateToNote={() => navigateToNote(note.document.path)}
+                >
+                  <Badge
+                    variant="outline"
+                    key={note.document.path}
+                    className="text-xs max-w-40 text-muted hover:cursor-pointer hover:bg-interactive-hover"
+                  >
+                    <span className="text-ellipsis overflow-hidden whitespace-nowrap">
+                      {note.document.title}
+                    </span>
+                  </Badge>
+                </RelevantNotePopover>
+              ))}
+            </CardDescription>
+          )}
+        </CardHeader>
+        <CollapsibleContent>
+          <CardContent className="px-4 pb-2 pt-0 max-h-96 overflow-y-auto flex flex-col gap-2">
+            {relevantNotes.map((note) => (
+              <RelevantNote
+                showPath={!inSameFolder(activeFile?.path ?? "", note.document.path)}
```

Comment:
We will only show the note path when the note is not in the same folder as the active note.